minio/signer.py

```diff
@@ -59,7 +59,7 @@ def _get_canonical_headers(headers):
 
     canonical_headers = {}
     for key, values in headers.items():
-        key = key.lower()
+        key = str(key).lower()
```

Comment:
As you have converted non-string to string via genheaders() -> normalize_headers() -> _metadata_to_headers(), there is no need to do it again.

Reply:
This is to specifically fix a scenario when `kv = dict(); kv['1'] = 'value'`  - I saw that this is not supported @balamurugana